app/components/alchemy/admin/resource/table.rb

```diff
@@ -0,0 +1,148 @@
+          :render_icon,
+          :link_to_dialog,
+          :edit_resource_path,
+          :resource_handler,
+          :resource_window_size,
+          to: :helpers
+
+        attr_reader :collection,
+          :nothing_found_label,
+          :search_filter_params
+
+        renders_many :headers, Header
+
+        renders_many :cells, ->(css_classes, &block) do
+          Cell.new(css_classes, &block)
+        end
+
+        renders_many :actions, ->(name, tooltip = nil, &block) do
+          Action.new(name, tooltip, &block)
+        end
+
+        erb_template <<~ERB
+          <% if collection.any? %>
+            <table class="list">
+              <thead>
+                <tr>
+                  <% headers.each do |header| %>
+                    <%= header %>
+                  <% end %>
+                  <% if actions? %>
+                    <th class="tools"></th>
+                  <% end %>
+                </tr>
+              </thead>
+              <tbody>
+                <% collection.each do |resource| %>
+                  <tr class="<%= cycle('even', 'odd') %>">
+                    <% cells.each do |cell| %>
+                       <%= render cell.with_resource(resource) %>
+                    <% end %>
+                    <% if actions? %>
+                      <td class="tools">
+                        <% actions.each do |action| %>
+                          <%= render action.with_resource(resource) %>
+                        <% end %>
+                      </td>
+                    <% end %>
+                  </tr>
+                <% end %>
+              </tbody>
+            </table>
+          <% else %>
+            <alchemy-message type="info">
+              <%= nothing_found_label %>
+            </alchemy-message>
+          <% end %>
+        ERB
+
+        def initialize(collection, query: nil, nothing_found_label: Alchemy.t("Nothing found"), search_filter_params: {})
+          @collection = collection
+          @query = query
+          @nothing_found_label = nothing_found_label
+          @search_filter_params = search_filter_params
+        end
+
+        def column(name, label: nil, sortable: false, type: nil, alignment: nil, &block)
+          label ||= resource_handler.model.human_attribute_name(name)
+          type ||= resource_handler.model.columns_hash[name.to_s]&.type
+          attribute = resource_handler.attributes.find { |item| item[:name] == name.to_s } || {name: name, type: type}
+          block ||= lambda { |item| render_attribute(item, attribute) }
+
+          css_classes = [name, type, alignment].compact.join(" ")
+          with_header(name, @query, css_classes: css_classes, label: label, type: type, sortable: sortable)
+          with_cell(css_classes, &block)
+        end
+
+        def icon_column(icon = nil, style: nil)
+          column(:icon, label: "") do |resource|
+            render_icon(icon || yield(resource), size: "xl", style: style)
```

Comment:
What's `resource` here? And what does `yield(resource)` pass to `render_icon` if no `icon` is present?

Reply:
The sites module now misses it's icon

### Main branch

<img width="356" alt="Monosnap 2024-08-27 18-13-57" src="https://github.com/user-attachments/assets/a7a72801-3b55-4d6c-bcbc-c4d19f803903">

### This PR

<img width="395" alt="Monosnap 2024-08-27 18-14-22" src="https://github.com/user-attachments/assets/a4a2690a-d52e-46b0-bf9b-7f55c4c57798">

Reply:
This is now fixed in the website view. This icon column can be used as a string or as a block and the `resource` is the object of the current table row.